File: FrontEnd/GranjaSA/src/app/main/Domain/Alimentacion.ts
export class Alimentacion {
    id?: number;
    descripcion: string;
    dosis: string;

    constructor(id: number | undefined = undefined, descripcion: string, dosis: string) {
        this.id = id;
        this.descripcion = descripcion;
        this.dosis = dosis;
    }
}

File: BackEnd/src/main/java/com/example/granjasa/Entity/Alimentacion.java
package com.example.granjasa.Entity;

import java.io.Serializable;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

@Entity
@Data
@Table(name = "alimentacion")
public class Alimentacion implements Serializable {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @NotNull(message = "La descripción es obligatoria")
    @Column(nullable = false)
    private String descripcion;

    @NotNull(message = "La dosis es requerida")
    @Column(nullable = false)
    private String dosis;

}
